### ISSUE TITLE
I need to gather data from 5 to 100 offset with limit 1

### ISSUE THREAD

**Issue by takielias** [aerni/laravel-spotify#28]
I need to gather data from 5 to 100 offset as limit 1. It should rate limit friendly. Is there any suitable way to do this?

**Comment by aerni** [comment]
I would probably do one request with a high limit instead of multiple requests with a limit of 1. And then cache the response or do whatever with it and do further manipulations from that response.

**Comment by takielias** [comment]
> 
> 
> I would probably do one request with a high limit instead of multiple requests with a limit of 1. And then cache the response or do whatever with it and do further manipulations from that response.

Your idea is nice But It would be great If we can set the previously generated Access Token for a specific time period.

**Comment by aerni** [comment]
I'm not sure what one has to do with the other? Please elaborate more what you're trying to do and why.

**Comment by takielias** [comment]
> 
> 
> I'm not sure what one has to do with the other? Please elaborate more what you're trying to do and why.

Suppose I want to Search Playlist. So I will use `Spotify::searchPlaylists('rock')-limit(1)->offset(1)->get();`

````
GET /v1/search?q=rock&type=playlist&limit=1&offset=1 HTTP/1.1
Host: api.spotify.com
Connection: close
Authorization: Bearer
BQB0RXeRUtI-UpXmR6CmTt3L-Hhq_vQoNfUaX8kbrRxTM1yJBNVvs9Km7aMfBBf6lt7Q
Bg9lgayiqQiYAxhjA0I-6LMrJjYvXlF8eOkWh8othjd0agvfZ3vyl_XCFz_pkhdhqc5blaqcEIy
Z3xL8dHuxL32qGW8I1YtVHXIpuUXf
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like
Gecko) Chrome/87.0.4280.66 Safari/537.36
Accept: */*
Origin: https://developer.spotify.com
Sec-Fetch-Site: same-site
Sec-Fetch-Mode: cors
Sec-Fetch-Dest: empty
Referer: https://developer.spotify.com/
Accept-Encoding: gzip, deflate
Accept-Language: en-US,en;q=0.9
````
But please check the above request we just need **Bearer** token to make that request. So If we can set the previously generated Bearer token, we can reduce 1 HTTP request per API endpoint for 1 hour because the token is valid for 1 hour. Is not it?

**Comment by aerni** [comment]
That's a good point. I will take a look at it.

**Comment by aerni** [comment]
This has been implemented in v1.6.1.
